nixos/modules/services/misc/nix-daemon.nix

```diff
@@ -446,6 +445,8 @@ in
         if [ "$USER" != root -o ! -w /nix/var/nix/db ]; then
             export NIX_REMOTE=daemon
         fi
+      '' + ''
+        export NIX_PATH="$HOME/.nix-defexpr/channels''${NIX_PATH:+:$NIX_PATH}"
```

Comment:
Are these two single quotes necessary here?

Reply:
Within the double-single-quote they escape the dollar-sign (` '' ... ``${ .. ''`), if you're asking about the two single quotes in the middle of the line? Or are you suggesting dropping the outer `''` in favor of something else? I'm a fan of using `''` in most places for consistency if nothing else, but if there's a reason that's bad or there's a better alternative please LMK! :)

Reply:
I think adding a `test -e` here would also fix #40165.

We should probably either get this into 18.09 or revert https://github.com/NixOS/nixpkgs/pull/38351, this is going to cause a bunch of confusion otherwise.

Reply:
Agreed.  Not sure what you mean about 'test -e' but happy if you want to just add a commit doing so? :D

Reply:
Well nix shows a warning if NIX_PATH entries don't exist, so only extend it if ~/.nix-defexpr exists.

```bash
if [ -e "$HOME/.nix-defexpr/channels" ]; then
    export NIX_PATH="$HOME/.nix-defexpr/channels''${NIX_PATH:+:$NIX_PATH}"
fi
```

Reply:
Oh, right.  Thanks, updated!

Reply:
Unfortunately this means the user needs to log out before the NIX_PATH gets updated after they add their first channel, but this is an improvement.